src/Processor/Gotenberg.php

```diff
@@ -106,6 +106,9 @@ public function getPdfFromString(string $html, array $params = [], bool $returnF
         $tempFileName = uniqid('web2print_');
 
         $chromium = GotenbergAPI::chromium(\Pimcore\Config::getSystemConfiguration('gotenberg')['base_url']);
+        if (method_exists($chromium, 'pdf')) {
+            $chromium = $chromium->pdf();
+        }
```

Comment:
I'd change to this, there is no relevance in the order of userAgent getting set, so can move up there.
Also reducing the `method_exists()` as it's not optimal and adding some comments, so that would be easier to spot when eg. dropping support for v1 or for future BC breaks (v3+).

```suggestion
        // To support gotenberg-php v2 and so on
        if (method_exists($chromium, 'pdf')) {
            $chromium = $chromium->pdf();
        } else {
            // gotenberg-php v1 BC Layer for methods unsupported in v2
            if (isset($params['userAgent'])) {
               $chromium->userAgent($params['userAgent']);
            }
            if (isset($params['pdfFormat'])) {
               $chromium->pdfFormat($params['pdfFormat']);
            }
        }
```